unittests/allplatformstests.py

```diff
@@ -3029,6 +3029,26 @@ def test_clang_tidy(self):
         self.assertIn('cttest.cpp:4:20', out)
         self.assertNotIn(dummydir, out)
 
+    @skipIfNoExecutable('clang-tidy')
+    @unittest.skipIf(not is_git_repo(), 'Skipping because this is not in git repo')
+    def test_clang_tidy_fix(self):
```

Comment:
This seems terribly wrong. Look at how we test clang-format a few lines up.

```
        # Ensure that test project is in git even when running meson from tarball.
```

Reply:
Good catch, I will submit a new PR to make the test run in a a temp git repo.

Reply:
Thanks. :)

Reply:
Submitted to https://github.com/mesonbuild/meson/pull/12301